sdk/python/feast/infra/online_stores/astra.py

```diff
@@ -0,0 +1,301 @@
+                                                     "BLOB",
+                                                     "timestamp",
+                                                     "timestamp"
+                                                 ]
+                                                 )
+            self.session.execute(cql_create_table)
+
+            # Now create Index
+            cql_index = _create_index_cql(key_space, table_name+"_ek", table_name, "entity_key")
+            self.session.execute(cql_index)
+
+        for table in tables_to_delete:
+            table_name = _table_id(project, table)
+            delete_cql = _create_delete_table_cql(key_space, table_name)
+            self.session.execute(delete_cql)
+
+    def teardown(
+        self,
+        config: RepoConfig,
+        tables: Sequence[Union[FeatureTable, FeatureView]],
+        entities: Sequence[Entity],
+    ):
+        online_config = config.online_store
+        assert isinstance(online_config, AstraConfig)
+        key_space = online_config.keyspace
+        for table in tables:
+            table_name = _table_id(config.project, table)
+            cql_delete = _create_delete_table_cql(key_space, table_name)
+            self.session.execute(cql_delete)
+
+
+def _create_cql_table(key_space: str,
+                      table_name: str,
+                      primary_key: List,
+                      columns: List[str],
+                      column_types: List[str]
+                      ) -> str:
+    """
+    in this function we will create a CQL to create a table
+    """
+    assert len(columns) > 0, "Columns can not be empty in a Table. "
+
+    assert len(columns) == len(column_types), "Length of columns and type of columns length missmatch."
+
+    # If we have same amount of columns and their types
+    cql_create_table = "CREATE TABLE  IF NOT EXISTS " + key_space + "." + table_name
+    cql_create_table += " ("
+    for col, typ in zip(columns, column_types):
+        if col in primary_key:
+            cql_create_table += col + " " + typ + ", "
+        else:
+            cql_create_table += col + " " + typ + " , "
+    cql_create_table = cql_create_table[:-2]
+    cql_create_table += ", PRIMARY KEY (" + ", ".join(primary_key) + ") "
+    cql_create_table += ");"
+    return cql_create_table
+
+
+def _create_cql_insert_record(key_space: str,
+                              table_name: str,
+                              column_names: List[str],
+                              values: List[str]
+                              ) -> str:
+    """
+    This is general function to insert a record in table
+    """
+    assert len(column_names) > 0, "Columns are empty."
+
+    assert len(column_names) == len(values), " Values size and column names size are not equal."
+
+    cql_insert_record = "INSERT INTO " + key_space + "." + table_name + " ("
+    cql_insert_record += ", ".join(column_names) + ") VALUES ("
+    for value in values:

```

Comment:
Maybe we could use `prepare` here to session handle input values types... WDYT?

Reply:
yes, you are right i will change it